Sprint_2

LununPavel opened the pull request:
Изменил структуру класса BooksCollector по согласованию с наставником (Сергеем), так как код представленный в тренажере отличается от кода из git

Написал тесты для класса BooksCollector

LununPavel committed:
Изменил структуру класса BooksCollector по согласованию с наставником (Сергеем), так как код представленный в тренажере отличается от кода из git

Написал тесты для класса BooksCollector

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,128 @@ def test_add_new_book_add_two_books(self):
         # проверяем, что добавилось именно две
         # словарь books_rating, который нам возвращает метод get_books_rating, имеет длину 2
-        assert len(collector.get_books_rating()) == 2
+        assert len(collector.books_genre) == 2
```
Нужно исправить: для этого есть свой метод get_books_genre. Тестировать нужно программу а не питон

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,128 @@ def test_add_new_book_add_two_books(self):
+    def test_get_books_with_specific_genre_get_book_names_with_same_genre(self):
+        collector = BooksCollector()
+
+        collector.books_genre = {
+            'Сияние': 'Ужасы',
+            'Шаровая молния': 'Фантастика',
+            'Остров фантазий': 'Ужасы',
+            'Шрек': 'Мультфильмы',
+            'Солнцестояние': 'Ужасы'
+        }
```
Нужно исправить: это следует делать не напрямую создавая словарь объекта а методами программы

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,128 @@ def test_add_new_book_add_two_books(self):
+    def test_get_books_with_specific_genre_get_empty_list_of_books(self):
+        collector = BooksCollector()
+
+        collector.books_genre = {
+            'Сияние': 'Ужасы',
+            'Шаровая молния': 'Фантастика',
+            'Остров фантазий': 'Ужасы',
+            'Шрек': 'Мультфильмы',
+            'Солнцестояние': 'Ужасы'
+        }
```
comment up

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,128 @@ def test_add_new_book_add_two_books(self):
+        collector.add_new_book('Бессмертие')
+        collector.add_new_book('Скрытые намерения')
+
+        for i in collector.books_genre:
+            collector.add_book_in_favorites(i)
```
Нужно исправить: for в тестах - не самое хорошее решение

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,128 @@ def test_add_new_book_add_two_books(self):
+        assert collector.favorites == ['Бессмертие', 'Ящик Скиннера', 'Забытая девушка', 'Скрытые намерения']
+
+
+    def test_delete_book_from_favorites_get_list_of_favorite_books(self):
+        collector = BooksCollector()
+
+        collector.favorites = ['Бессмертие', 'Ящик Скиннера', 'Забытая девушка', 'Скрытые намерения']
+        collector.delete_book_from_favorites('Забытая девушка')
+
+        assert collector.favorites == ['Бессмертие', 'Ящик Скиннера', 'Скрытые намерения']
+
+
+    def test_get_list_of_favorites_books_get_list_of_two_selected_books(self):
+        collector = BooksCollector()
+
+        collector.favorites = ['Бессмертие','Забытая девушка', 'Скрытые намерения']
+        collector.delete_book_from_favorites('Забытая девушка')
+
+        assert collector.get_list_of_favorites_books() == ['Бессмертие', 'Скрытые намерения']
```
Нужно исправить: два одинаковых по сути теста

LununPavel committed:
Исправление замечаний:

1. Во всех тестах изменил проверки на методы программы для тестирования самой программы, а не питона
2. Заменил напрямую созданные словари и списки объектов на метода программы
3. Убрал цикл в методе test_add_book_in_favorites_add_three_books_in_favorites
4. Переписал два последних теста во избежании дублирования тестов

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,161 @@ def test_add_new_book_add_two_books(self):
+    def test_get_books_with_specific_genre_get_empty_list_of_books(self):
+        collector = BooksCollector()
+
+        collector.add_new_book('Сияние')
+        collector.set_book_genre('Сияние', 'Ужасы')
+        collector.add_new_book('Шаровая молния')
+        collector.set_book_genre('Шаровая молния', 'Фантастика')
+        collector.add_new_book('Остров фантазий')
+        collector.set_book_genre('Остров фантазий', 'Ужасы')
+        collector.add_new_book('Шрек')
+        collector.set_book_genre('Шрек', 'Мультфильмы')
+        collector.add_new_book('Солнцестояние')
+        collector.set_book_genre('Солнцестояние', 'Ужасы')
```
Можно лучше: многовато действия чтобы проверить что нет такого жанра, это юниты, они должны быть минимальными, быстрыми, информативными

AlexKitIt commented on tests.py:
```diff
@@ -18,7 +19,161 @@ def test_add_new_book_add_two_books(self):
+        collector.delete_book_from_favorites('Забытая девушка')
+        collector.delete_book_from_favorites('Бессмертие')
+
+        assert ['Бессмертие'] and ['Скрытые намерения'] not in collector.get_list_of_favorites_books()
```
Нужно исправить: collector.get_list_of_favorites_books() вернет список. То, что в нем не будет подсписков ['Бессмертие'] and ['Скрытые намерения'] это в любом случае. Тест не функционален

LununPavel committed:
Исправление замечаний:

1. Сделал меньше входные данные в тестах для информативности и быстроты выполнения
2. Изменил проверку в тесте удаления книги из списка избранных. В сам список добавляю две книги и методом удаляю одну, а в проверке смотрю, что в списке находится одна книга
